### DISCUSSION TITLE
Can I scroll with mouse to navigate to next/previous page (drun modi)

### DISCUSSION THREAD

**Discussion by xlucn** [davatorium/rofi#1858] [Q&A] [answered]
I would like to scroll to next or previous page with mouse wheel, for example in
```
rofi -show drun
```
I am not sure if this could be done with current rofi configurations. The current options for mouse bindings, according to the manual, are very limited:
   ml-row-left,
   ml-row-right,
   ml-row-up,
   ml-row-down,
   me-select-entry,
   me-accept-entry,
   me-accept-custom

I have opened an feature request before (#1755), which I think would achieve this behavior. What does the dev think of this feature as a solution?

**Comment by DaveDavenport** [comment]
I think something more flexible would be nice, but I have not had time to look at it. 
Don't use the mouse myself.

A patch is welcome.

**Reply by DaveDavenport** [reply, marked as answer]
Talked to @sardemff7  who wrote the key handling part.  This already works.

```bash
rofi -kb-page-prev ScrollUp  -ml-row-up '' -show keys
```

Unset the current binding (that only works when hovering over the listview) and set the global one to ScrollUp. This works.

**Reply by xlucn** [reply]
That's great news (which is not new)! I never thought of actually trying that. I will mark this as answer and probably close that feature request issue then.

However, this behavior is now confusing. E.g. the two behaves the same:
```
    kb-row-up:            "Up,k,ScrollUp";
    ml-row-up:            "Up,k,ScrollUp";
```

**Reply by DaveDavenport** [reply]
No, `ml-` only works when hovering over the listview. `kb-` is global.

**Reply by xlucn** [reply]
Indeed, that makes sense now.
